extensions/iceberg/src/main/java/io/deephaven/iceberg/location/IcebergTableParquetLocation.java

```diff
@@ -0,0 +1,86 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.location;
+
+import io.deephaven.api.ColumnName;
+import io.deephaven.api.SortColumn;
+import io.deephaven.engine.table.impl.locations.TableKey;
+import io.deephaven.engine.table.impl.locations.TableLocation;
+import io.deephaven.iceberg.util.IcebergTableAdapter;
+import io.deephaven.parquet.table.ParquetInstructions;
+import io.deephaven.parquet.table.location.ParquetTableLocation;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.NullOrder;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SortDirection;
+import org.apache.iceberg.SortField;
+import org.apache.iceberg.SortOrder;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class IcebergTableParquetLocation extends ParquetTableLocation implements TableLocation {
+
+    @Nullable
+    private final List<SortColumn> sortedColumns;
+
+    public IcebergTableParquetLocation(
+            @NotNull final IcebergTableAdapter tableAdapter,
+            @NotNull final TableKey tableKey,
+            @NotNull final IcebergTableParquetLocationKey tableLocationKey,
+            @NotNull final ParquetInstructions readInstructions) {
+        super(tableKey, tableLocationKey, readInstructions);
+        sortedColumns = computeSortedColumns(tableAdapter, tableLocationKey.dataFile());
+    }
+
+    @Override
+    @NotNull
+    public List<SortColumn> getSortedColumns() {
+        return sortedColumns == null ? super.getSortedColumns() : sortedColumns;
+    }
+
+    @Nullable
+    private static List<SortColumn> computeSortedColumns(
+            @NotNull final IcebergTableAdapter tableAdapter,
+            @NotNull final DataFile dataFile) {
+        final Integer sortOrderId = dataFile.sortOrderId();
```

Comment:
Need to think about behavior when unsorted (either b/c null or explicitly set to unsorted)...